snaky.js

```diff
@@ -87,19 +108,21 @@ function moveSnake() {
     case "up":
       head = {x: snake[0].x, y: snake[0].y - dy};
       break;
-  }
 
+    }
+  
   snake.unshift(head);
   snake.pop();
 
-  
 }
 
 function foodEaten() {
+
   if (snake[0].x == apple[0].x && snake[0].y == apple[0].y) {
 
       apple[0].x = Math.floor(Math.random() * snakeboard.width / 10) * 10;
       apple[0].y = Math.floor(Math.random() * snakeboard.height / 10) * 10;
+      snake.push({x: snake[snake.length - 1].x + dx, y: snake[snake.length - 1].y + dy});
```

Comment:
It might be cleaner to re-implement snake as a class and write a grow() method on it or something similar. Mutating the snake array like this could get confusing and hard to manage.

Also, if you look carefully, on the frame that the snake eats the food, the segment of the body that gets added looks like it's a bit off. I think what you want to do is this:
```suggestion
      snake.push({x: snake[snake.length - 1].x, y: snake[snake.length - 1].y});
```